tests/web/test_db_query.py

```diff
@@ -714,3 +719,56 @@ def test_db_query_ignore_consistent() -> None:
     )
     assert result.extra["stats"]["consistent"] is False
     assert result.extra["stats"]["max_threads"] == 10
+
+
+@pytest.mark.redis_db
+@pytest.mark.clickhouse_db
+@pytest.mark.parametrize(
+    "disable_lua_randomize_query_id, disable_lua_scripts_sample_rate, expected_startswith, test_cache_hit_simple",
+    [
+        (0, 0, "test_query_id", False),
+        (1, 1, "randomized-", True),
+    ],
+)
+def test_clickhouse_settings_applied_to_query_id(
+    disable_lua_randomize_query_id: int,
+    disable_lua_scripts_sample_rate: int,
+    expected_startswith: str,
+    test_cache_hit_simple: bool,
+) -> None:
+    query, storage, attribution_info = _build_test_query("count(distinct(project_id))")
+    state.set_config("disable_lua_randomize_query_id", disable_lua_randomize_query_id)
+    state.set_config(
+        "read_through_cache.disable_lua_scripts_sample_rate",
+        disable_lua_scripts_sample_rate,
+    )
+
+    query_settings = HTTPQuerySettings()
+    formatted_query = format_query(query)
+    reader = storage.get_cluster().get_reader()
+    clickhouse_query_settings: Dict[str, Any] = {}
+    stats: dict[str, Any] = {}
+
+    with mock.patch(
+        "snuba.web.db_query.get_query_cache_key", return_value="test_query_id"
```

Comment:
why does this need to be mocked? The cache key should be the same if the query is the same

Reply:
Reason for mocking, is in the test query_id is not computed.